src/Sarif.Viewer.VisualStudio.ResultSources.DeveloperCanvas.Core/Services/DevCanvasWebAPIAccessor.cs

```diff
@@ -0,0 +1,185 @@
+                }
+            }
+
+            return generatorList;
+        }
+
+        /// <inheritdoc/>
+        private async Task<List<DevCanvasGeneratorInfo>> TryGetGeneratorsFromWebApiAsync()
+        {
+            string sarifUrl = $"https://{currentServer}/api/v{version}/SarifInsight/SarifInsightProviders";
+            HttpClient client = await authManager.GetHttpClientAsync();
+            if (client != null)
+            {
+                try
+                {
+                    string responseBody;
+                    using (HttpResponseMessage response = await client.GetAsync(sarifUrl))
+                    {
+                        response.EnsureSuccessStatusCode();
+                        responseBody = await response.Content.ReadAsStringAsync();
+                    }
+                    return JsonConvert.DeserializeObject<List<DevCanvasGeneratorInfo>>(responseBody); // get generators here
+                }
+                catch (Exception e)
+                {
+                    Trace.WriteLine(e);
+                    return new List<DevCanvasGeneratorInfo>();
+                }
+            }
+            else
+            {
+                Trace.WriteLine($"Failed to access {currentServer} endpoint with supplied credentials.");
+            }
+            return new List<DevCanvasGeneratorInfo>();
+        }
+
+
+        public async Task<SarifLog> GetSarifLogV1Async(DevCanvasRequestV1 request)
+        {
+            string Url = $"https://{currentServer}/api/v{version}/SarifInsight/SarifInsightsForFile";
+
+            HttpClient client = await authManager.GetHttpClientAsync();
+            if (client != null)
+            {
+                string requestJson = JsonConvert.SerializeObject(request);
+                StringContent content = new StringContent(requestJson, System.Text.Encoding.UTF8, "application/json");
+                HttpResponseMessage response = null;
+                try
+                {
+                    Stopwatch stopwatch = Stopwatch.StartNew();
+                    using (response = await client.PostAsync(Url, content))
+                    {
+                        response.EnsureSuccessStatusCode();
+                        stopwatch.Stop();
+                        string responseBody = await response.Content.ReadAsStringAsync();
+                        SarifLog log = JsonConvert.DeserializeObject<SarifLog>(responseBody);
+                        int resultCount = log.Runs[0].Results.Count;
+                        Trace.WriteLine($"Took {stopwatch.ElapsedMilliseconds}ms to query for {resultCount} insights.");
+                        return log;
+                    }
+                }
+                catch (Exception e)
+                {
+                    // we want to swallow and return an empty list
+                    Trace.WriteLine($"Failed to access {currentServer} endpoint.\nReceived error code {response?.StatusCode}.\nException: {e}");
```

Comment:
Do we want to track time still down failure paths?  Could help pinpoint failures on client versus server side.